config/webpackDevServer.config.cjs

```diff
@@ -0,0 +1,49 @@
+const ignoredFiles = require('react-dev-utils/ignoredFiles');
+const paths = require('./paths.cjs');
+
+const host = process.env.HOST || '0.0.0.0';
+const sockHost = process.env.WDS_SOCKET_HOST;
+const sockPath = process.env.WDS_SOCKET_PATH; // default: '/ws'
+const sockPort = process.env.WDS_SOCKET_PORT;
+
+module.exports = function (proxy, allowedHost) {
+  const disableFirewall =
+    !proxy || process.env.DANGEROUSLY_DISABLE_HOST_CHECK === 'true';
+  return {
+    allowedHosts: disableFirewall ? 'all' : [allowedHost],
```

Comment:
Also curious as to what this block does as well.

Reply:
This config for the webpack (Dev Server) which is what allows you to run this locally and have hot reloading and such was basically all boiler code that CRA Eject spit out for me I configured most of what I knew was needed and left the rest (Note that this config is only for the dev environment hence why its only a dev dependency)

Chat GPT 4 response more detailed:

> Setting DANGEROUSLY_DISABLE_HOST_CHECK to 'true' and disabling the host check can have significant security implications, especially in production environments. It opens your server to potential DNS rebinding attacks, where an attacker could bypass the same-origin policy, which in turn could lead to cross-site request forgery (CSRF), cross-site scripting (XSS), and other vulnerabilities.
> The configuration is likely intended for development purposes, where ease of testing (e.g., from various hosts or through proxies) might necessitate a more permissive setup. However, care should be taken to ensure that such settings are not used in a production environment.
> Use Case:
> 
> This setup could be useful in development environments where developers are working across different network configurations or need to test the application from various hosts. It allows for flexibility in connecting to the webpack dev server but should be used with caution due to the security considerations mentioned.